vector/src/main/java/im/vector/activity/VectorRoomActivity.java

```diff
@@ -3859,7 +3849,13 @@ public void onNetworkError(Exception e) {
 
                             @Override
                             public void onMatrixError(MatrixError e) {
-                                onError(e.getLocalizedMessage());
+                                // Catch here the consent request if any.
+                                if (MatrixError.M_CONSENT_NOT_GIVEN.equals(e.errcode)) {
+                                    hideWaitingView();
+                                    getConsentNotGivenHelper().displayDialog(e);
```

Comment:
Maybe call `onConsentNotGiven(null, e);` here, instead of `getConsentNotGivenHelper().displayDialog(e);`?